### PR TITLE
Change Option.None to undefined

### LINKED ISSUE #2028: Null is erroneously generated instead of undefined for optional arguments
```ts
		/**
		 * Create a new position relative to this position.
		 *
		 * @param lineDelta Delta value for the line value, default is `0`.
		 * @param characterDelta Delta value for the character value, default is `0`.
		 * @return A position which line and character is the sum of the current line and
		 * character and the corresponding deltas.
		 */
		translate(lineDelta?: number, characterDelta?: number): Position;
```

This is from the VSCode API. Note that the type of the first argument `lineDelta` is `number | undefined`. This is what the editor tells me when I hover over it.

```fs
    /// <summary>Create a new position relative to this position.</summary>
    /// <param name="lineDelta">Delta value for the line value, default is `0`.</param>
    /// <param name="characterDelta">Delta value for the character value, default is `0`.</param>
    abstract translate: ?lineDelta: int * ?characterDelta: int -> Position
```

Here is the binding generated by `ts2fable`. I've changed `float` to `int` by hand here.

```fs
let t = startPosition.translate(characterDelta=x.Length)
```

This is how I use it in my code. I am omitting the first argument on purpose.

```js
const t = startPosition.translate(null, x$$6[0].length);
```

This is the generated code for it. Note the `null` in the first argument. When I step through it with a debugger, I get an exception raised.

```ts
	translate(change: { lineDelta?: number; characterDelta?: number; }): Position;
	translate(lineDelta?: number, characterDelta?: number): Position;
	translate(lineDeltaOrChange: number | undefined | { lineDelta?: number; characterDelta?: number; }, characterDelta: number = 0): Position {

		if (lineDeltaOrChange === null || characterDelta === null) {
			throw illegalArgument();
		}
``` 

This is how it is implemented. I omitted the extra overload you see here in the previous description.

### PR DESCRIPTION
**Pull request by ncave**
- Fixes #2028.

### CODE REVIEW

**Comment by ncave** [comment]
>@Zaid-Ajaj commented:
Alright so might be a huge breaking change. null | T was a default for Option<T> for quite some time. I would say if people wanted a different behaviour then it is simply by writing a function that converts T -> T and null -> undefined to get that behaviour.

@Zaid-Ajaj I agree, we should merge this as part of Fable 3.0 beta, to avoid breaking existing code.
But, it should just work most of the time, and #2028 has a valid argument on optional parameters.

**Comment by alfonsogarciacaro** [comment]
In principle Fable tries to turn a deaf ear into the null/undefined difference in JS and every time there's a null check, the non-strict equality operator `==` is used which will give true both for null and undefined. There was no particular reason why `null` was chosen to represent to Option.None besides .NET influence, I guess if it improves interoperability with JS it can be changed and hopefully it doesn't break anything.

We can still use `null` to output F# `null` in Nullable types. But I'd be more cautious in going further to make the distinction in runtime (e.g. by using the strict equality operator `===` for Option.None checks).

**Comment by MangelMaxime** [comment]
For me, the main reason for this change is this one:

> By looking to [MDN](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Fonctions/Valeurs_par_d%C3%A9faut_des_arguments) and [this blog post](https://codeburst.io/javascript-null-vs-undefined-20f955215a2).
> 
> It indeed seems like we should emit `undefined` instead of `null`. This allows JavaScript to pick up the default value for example.

https://github.com/fable-compiler/Fable/issues/2028#issuecomment-616079794

But, yes we probably should not make the equality operator too strict as @alfonsogarciacaro said.

**Comment by ncave** [comment]
> But, yes we probably should not make the equality operator too strict as @alfonsogarciacaro said.

@MangelMaxime I agree, that's why all the comparisons in this PR are left as they were (non-strict).

File: src/fable-library/Util.ts [hunk 1]
```diff
@@ -671,25 +671,25 @@ export function partialApply(arity: number, f: Function, args: any[]): any {
         // Wrap arguments to make sure .concat doesn't destruct arrays. Example
         // [1,2].concat([3,4],5)   --> [1,2,3,4,5]    // fails
         // [1,2].concat([[3,4],5]) --> [1,2,[3,4],5]  // ok
-        return (a1: any) => f.apply(null, args.concat([a1]));
+        return (a1: any) => f.apply(undefined, args.concat([a1]));
```
Comment (alfonsogarciacaro) [inline src/fable-library/Util.ts:674]: Just aesthetics, but for `apply` it seems either `null` or `undefined` will work and the examples in MDN use `null`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/apply

**Comment by alfonsogarciacaro** [comment]
I will test the branch with a couple of my projects to check there are no issues :+1:

File: src/fable-library/Util.ts (hunk 1 above)
Comment (ncave) [reply src/fable-library/Util.ts:674]: @alfonsogarciacaro Yes, the second note in the MDN article says `When the first argument is undefined or null ...`

**Comment by alfonsogarciacaro** [comment]
Sorry, I totally forgot about this! 🤦 I just tested with one project and everything seems to be working fine, so I'm OK with merging the PR :+1:

**Pull request merged**